rxlifecycle/src/test/java/com/trello/rxlifecycle/UntilCorrespondingEventSingleTransformerTest.java

```diff
@@ -5,66 +5,74 @@
 import rx.Single;
 import rx.functions.Func1;
 import rx.observers.TestSubscriber;
+import rx.schedulers.TestScheduler;
 import rx.subjects.PublishSubject;
 
 import java.util.concurrent.CancellationException;
+import java.util.concurrent.TimeUnit;
 
 public class UntilCorrespondingEventSingleTransformerTest {
 
     PublishSubject<String> lifecycle;
     TestSubscriber<String> testSubscriber;
+    TestScheduler testScheduler; // Since Single is not backpressure aware, use this to simulate it taking time
 
     @Before
     public void setup() {
         lifecycle = PublishSubject.create();
-        testSubscriber = new TestSubscriber<>(0);
+        testSubscriber = new TestSubscriber<>();
+        testScheduler = new TestScheduler();
     }
 
     @Test
     public void noEvents() {
         Single.just("1")
+            .delay(1, TimeUnit.MILLISECONDS, testScheduler)
             .compose(new UntilCorrespondingEventSingleTransformer<String, String>(lifecycle, CORRESPONDING_EVENTS))
             .subscribe(testSubscriber);
 
-        testSubscriber.requestMore(1);
+        testScheduler.advanceTimeBy(1, TimeUnit.MILLISECONDS);
```

Comment:
Also in my opinion it increases readability of the test and the purpose becomes a bit clearer of what is being tested.